### PR TITLE
Tsm 85 large backup s3

### PR DESCRIPTION
**Pull request by the-wtag**
# Background
Large File Upload (files>5GB) is not possible with existing implementation. Have to fix this issue and allow large files to be uploaded.

# Changes
Use `MultipartUploader` instead of  `putObject` as Multipart Upload allow files upload regardless of size limitation.
_Refs: https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/s3-multipart-upload.html_

# Testing Instruction:
Integra PR(`integra/pull/214`) is set to check if the command(That is run in a scheduled manner in origin setup) is successfully backing up necessary things. To test this 
- Login to pr env using `platform ssh -e pr-214`
- Run backup Command `vendor/bin/psh-toolbelt backup:branch --force`
- Check if backup is stored in aws bucket using `aws s3 ls s3://wndrs-backups-test --recursive` (or any relevant command) and also configure `aws` client with credentials from Jira Ticket

### CODE REVIEW

**Comment by the-wtag** [comment]
Regarding TSM-85, Travis is checking using composer v.2 .But my new code pushes was not passing Travis indicating some package version mismatch. So i ran `composer update` and `composer require --dev dealerdirect/phpcodesniffer-composer-installer:"^0.7.0"` (Following this https://github.com/Dealerdirect/phpcodesniffer-composer-installer/issues/119) . But this time composer codestyle is throwing error ( Even if i do same thing with master )
```
hemel@Jontor-Montor:~/Desktop/DrupalWork/Wondrous/devfol/psh-toolbelt$ git stash
No local changes to save
hemel@Jontor-Montor:~/Desktop/DrupalWork/Wondrous/devfol/psh-toolbelt$ git checkout master
Switched to branch 'master'
Your branch is up to date with 'origin/master'.
hemel@Jontor-Montor:~/Desktop/DrupalWork/Wondrous/devfol/psh-toolbelt$ composer codestyle
> ./vendor/bin/phpcs
E....E.....E 12 / 12 (100%)



FILE: src/Commands/TestCommands.php
-------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-------------------------------------------------------------------------------------------------------------------------------------------------------------
 8 | ERROR | [x] Non-namespaced classes/interfaces/traits should not be referenced with use statements
   |       |     (Drupal.Classes.UseGlobalClass.RedundantUseStatement)
-------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: src/SiteSettings.php
--------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
--------------------------------------------------------------------------------------------------------------------------------------------------------------
  7 | ERROR | [x] Non-namespaced classes/interfaces/traits should not be referenced with use statements
    |       |     (Drupal.Classes.UseGlobalClass.RedundantUseStatement)
 10 | ERROR | [x] Non-namespaced classes/interfaces/traits should not be referenced with use statements
    |       |     (Drupal.Classes.UseGlobalClass.RedundantUseStatement)
 11 | ERROR | [x] Non-namespaced classes/interfaces/traits should not be referenced with use statements
    |       |     (Drupal.Classes.UseGlobalClass.RedundantUseStatement)
--------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------------------------------------------------------------------


FILE: src/Commands/BackupCommands.php
--------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 3 ERRORS AFFECTING 3 LINES
--------------------------------------------------------------------------------------------------------------------------------------------------------------
  8 | ERROR | [x] Non-namespaced classes/interfaces/traits should not be referenced with use statements
    |       |     (Drupal.Classes.UseGlobalClass.RedundantUseStatement)
  9 | ERROR | [x] Non-namespaced classes/interfaces/traits should not be referenced with use statements
    |       |     (Drupal.Classes.UseGlobalClass.RedundantUseStatement)
 11 | ERROR | [x] Non-namespaced classes/interfaces/traits should not be referenced with use statements
    |       |     (Drupal.Classes.UseGlobalClass.RedundantUseStatement)
--------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 3 MARKED SNIFF VIOLATIONS AUTOMATICALLY
--------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 137ms; Memory: 8MB

Script ./vendor/bin/phpcs handling the codestyle event returned with error code 2
```

**Comment by yra-wtag** [comment]
It looks overall good to me. If Rainer suggests us to follow the new guideline for coding style then @the-wtag please fix the lint issues

**Review by yra-wtag — APPROVED** [review]

File: composer.lock [hunk 1]
```diff
@@ -5467,20 +6645,24 @@
...
+            "support": {
+                "issues": "https://github.com/phpstan/phpdoc-parser/issues",
+                "source": "https://github.com/phpstan/phpdoc-parser/tree/master"
+            },
             "time": "2019-06-07T19:13:52+00:00"
         },
         {
             "name": "phpstan/phpstan",
-            "version": "0.11.19",
+            "version": "0.11.20",
```
Comment (rainer-friederich) [inline composer.lock:6656]: You propably get the new sniffs from this update. So if you update phpstan you also need to fix the sniffs

**Review by rainer-friederich — CHANGES_REQUESTED** [review]

File: src/Commands/BackupCommands.php [hunk 2]
```diff
@@ -288,13 +294,11 @@ private function dbDumpAndUpload(string $prefix) : void {
       $drushPath = Robo::config()->get('drush.path');
```
Comment (zar-wtag) [inline src/Commands/BackupCommands.php:291]:
```suggestion
      $robo = new \Robo\Robo();
      $drushPath = $robo->config()->get('drush.path');
```

File: src/Commands/BackupCommands.php [hunk 3]
```diff
@@ -288,13 +294,11 @@ private function dbDumpAndUpload(string $prefix) : void {
...
 
-      $this->s3Client->putObject(
-            [
-              'Bucket' => Robo::config()->get('storage.s3.upload_bucket'),
-              'Key' => $objectKey,
-              'Body' => fopen($pathToFile, 'r'),
-            ]
-        );
+      $this->multipartUploader = new MultipartUploader($this->s3Client, fopen($pathToFile, 'r'), [
+        'bucket' => Robo::config()->get('storage.s3.upload_bucket'),
```
Comment (zar-wtag) [inline src/Commands/BackupCommands.php:295]:
```suggestion
        'bucket' => $robo->config()->get('storage.s3.upload_bucket'),
```

File: src/Commands/BackupCommands.php [hunk 4]
```diff
@@ -358,14 +362,12 @@ private function archiveAndUploadFiles(string $prefix) : void {
...
-        $this->s3Client->putObject(
-              [
-                'Bucket' => Robo::config()->get('storage.s3.upload_bucket'),
-                'Key' => $objectKey,
-                'Body' => fopen($targetFile, 'r'),
-              ]
-          );
+        
+        $this->multipartUploader = new MultipartUploader($this->s3Client, fopen($targetFile, 'r'), [
+          'bucket' => Robo::config()->get('storage.s3.upload_bucket'),
```
Comment (zar-wtag) [inline src/Commands/BackupCommands.php:364]:
```suggestion
          'bucket' => $robo->config()->get('storage.s3.upload_bucket'),
```

**Review by zar-wtag — CHANGES_REQUESTED** [review]
Considering the coding standard changes I have suggested a code change that might be in accordance with the standard changes. Please review.

**Review by bas-wtag — APPROVED** [review]

**Comment by the-wtag** [comment]
TravisCI is green and Code is functional ( Tested from `integra` PR)

**Review by zar-wtag — APPROVED** [review]
Looks good to me. Code changes were explained.

**Review by rainer-friederich — APPROVED** [review]
LGTM. Please merge and tag a 2.1.0 release

**Review by MustafizurRahmanWelldev — APPROVED** [review]

**Pull request merged**